core/src/main/java/org/mule/el/context/MessageContext.java

```diff
@@ -100,14 +101,37 @@ public Object getPayload()
         }
     }
 
+    /**
+     * Obtains the payload of the current message transformed to the given #type.
+     * <p/>
+     * <b>Note:</b> If this current payload is a stream this operation will read the stream making it unreadable for
+     * further processing.  To avoid this transform the message prior to use of this expression.
+     * TODO MULE-9285
+     *
+     * @param type the java type the payload is to be transformed to
+     * @return the transformed payload
+     * @throws TransformerException
+     */
     public <T> T payloadAs(Class<T> type) throws TransformerException
```

Comment:
Yes, at least for now to maintain MEL compatibility. If we change this then it's part of a separate issue, we should decide if we want to expose transformation like this to the user in EL or via transformation service itself (for now transformation service is just being used as a way to refactor transformation out of the message)